src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './components/UserInfo';
+import { Todo } from './components/TodoInfo';
+import { newTodoId } from './helpers';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserIdValid, setIsUserIdValid] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [actualTodos, setTodos] = useState(todos);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const validTitle = title.trim();
+
+    const newId = newTodoId(todos);
+
+    if (validTitle) {
+      setIsTitleValid(true);
+    }
+
+    if (!userId) {
+      setIsUserIdValid(true);
+    }
+
+    if (!userId || validTitle) {
+      return;
+    }
+
+    const addedTodo: Todo = {
+      id: newId,
+      completed: false,
+      user: getUserById(Number(userId)),
+      title: validTitle,
+      userId: Number(userId),
```

Comment:
in `handleSelect` function you use `Number(event.target.value)`.
why do you need it here?🤔